app/server/src/main/scala/org/bitcoins/server/BitcoinSServerMain.scala

```diff
@@ -27,91 +22,17 @@ import org.bitcoins.wallet.Wallet
 import org.bitcoins.wallet.config.WalletAppConfig
 
 import scala.concurrent.{ExecutionContext, Future, Promise}
-import scala.util.Properties
-
-object Main extends App with HttpLogger {
-
-  def runMain(args: Vector[String]): Future[Unit] = {
-    val argsWithIndex = args.zipWithIndex
-
-    val dataDirIndexOpt = {
-      argsWithIndex.find(_._1.toLowerCase == "--datadir")
-    }
-    val datadirPathOpt = dataDirIndexOpt match {
-      case None => None
-      case Some((_, dataDirIndex)) =>
-        val str = args(dataDirIndex + 1)
-        val usableStr = str.replace("~", Properties.userHome)
-        Some(Paths.get(usableStr))
-    }
-
-    val configIndexOpt = {
-      argsWithIndex.find(_._1.toLowerCase == "--conf")
-    }
-    val baseConfig = configIndexOpt match {
-      case None =>
-        val configPath =
-          datadirPathOpt.getOrElse(AppConfig.DEFAULT_BITCOIN_S_DATADIR)
-        AppConfig
-          .getBaseConfig(configPath)
-          .resolve()
-      case Some((_, configIndex)) =>
-        val str = args(configIndex + 1)
-        val usableStr = str.replace("~", Properties.userHome)
-        val path = Paths.get(usableStr)
-        ConfigFactory
-          .parseFile(path.toFile)
-          .resolve()
-    }
-
-    val configDataDir = Paths.get(
-      baseConfig.getStringOrElse("bitcoin-s.datadir",
-                                 AppConfig.DEFAULT_BITCOIN_S_DATADIR.toString))
-    val datadirPath = datadirPathOpt.getOrElse(configDataDir)
-
-    val networkStr = baseConfig.getString("bitcoin-s.network")
-    val network = BitcoinNetworks.fromString(networkStr)
-
-    val datadir: Path = {
-      val lastDirname = network match {
-        case MainNet  => "mainnet"
-        case TestNet3 => "testnet3"
-        case RegTest  => "regtest"
-        case SigNet   => "signet"
-      }
-      datadirPath.resolve(lastDirname)
-    }
 
-    System.setProperty("bitcoins.log.location", datadir.toAbsolutePath.toString)
+class Main(override val args: Array[String]) extends BitcoinSRunner {
```

Comment:
I would really rather not call this or the object below `Main` as it starts to get confusing when searching in intellij. Maybe call this `BitcoinSServerMain`?

Reply:
you can always specify the `Main` that gets run in the build file here: https://github.com/bitcoin-s/bitcoin-s/blob/b1635a8f70b5a50cfd4c00c9781587cc0935127f/app/server/server.sbt#L9-L8